plutus-core/untyped-plutus-core/Language/UntypedPlutusCore/Core/Instance/Flat.hs

```diff
@@ -0,0 +1,57 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+{-# LANGUAGE LambdaCase           #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Language.UntypedPlutusCore.Core.Instance.Flat where
+
+import           Language.UntypedPlutusCore.Core.Type
+
+import           Language.PlutusCore.Flat
+import           Language.PlutusCore.Universe
+
+import           Data.Word                            (Word8)
+import           Flat
+import           Flat.Decoder
+import           Flat.Encoder
+
+encodeTerm :: Word8 -> Encoding
```

Comment:
`encodeConstructorTag`/`decodeConstructorTag`?

Reply:
Because different tags encode differently (types use 3 bits while terms use 4 bits, for example) I am using separate functions for each and every encoding.